src/omotes_simulator_core/entities/assets/pipe.py

```diff
@@ -15,55 +15,78 @@
-    _qheat_external: float
-    """The external heat flow into the pipe [W]."""
+    output: List[Dict[str, float]]
+    """The output list of the pipe with a dictionaries for each timestep."""
+
     length: float
     """The length of the pipe [m]."""
+
     diameter: float
-    """The diameter of the pipe [m]."""
+    """The inner diameter of the pipe [m]."""
+
     roughness: float
     """The wall roughness of the pipe [m]."""
+
     alpha_value: float
     """The alpha value of the pipe [W/(m2 K)]."""
-    output: List[Dict[str, float]]
-    """The output list of the pipe with a dictionaries for each timestep."""
 
-    def __init__(self, asset_name: str, asset_id: str, port_ids: list[str]):
+    minor_loss_coefficient: float
+    """The minor loss coefficient of the pipe [-]."""
+
+    external_temperature: float
+    """The external temperature surrounding the pipe [K]."""
+
+    qheat_external: float
+    """The external heat flow into the pipe [W]."""
+
+    name: str
+    """The name of the pipe."""
+
+    def __init__(
+        self,
+        asset_name: str,
+        asset_id: str,
+        port_ids: list[str],
+        length: float,
+        inner_diameter: float,
+        roughness: float,
+        alpha_value: float,
+        minor_loss_coefficient: float,
+        external_temperature: float,
+        qheat_external: float,
+    ):
         """Initialize a Pipe object.
 
         :param str asset_name: The name of the asset.
         :param str asset_id: The unique identifier of the asset.
         :param List[str] port_ids: List of ids of the connected ports.
+        :param float length: The length of the pipe [m].
+        :param float inner_diameter: The inner diameter of the pipe [m].
+        :param float minor_loss_coefficient: The minor loss coefficient of the pipe [-].
+        :param float external_temperature: The external temperature surrounding the pipe [K].
+        :param float qheat_external: The external heat flow into the pipe [W].
+        :param float roughness: The wall roughness of the pipe [m].
+        :param float alpha_value: The alpha value of the pipe [W/(m2 K)].
         """
         super().__init__(asset_name=asset_name, asset_id=asset_id, connected_ports=port_ids)
         # Initialize the default values of the pipe
-        self._minor_loss_coefficient = PIPE_DEFAULTS.minor_loss_coefficient
-        self._external_temperature = PIPE_DEFAULTS.external_temperature
-        self._qheat_external = PIPE_DEFAULTS.qheat_external
+        self.minor_loss_coefficient = minor_loss_coefficient
+        self.external_temperature = external_temperature
+        self.qheat_external = qheat_external
         # Define properties of the pipe
-        self.length = PIPE_DEFAULTS.length
-        self.diameter = PIPE_DEFAULTS.diameter
-        self.roughness = PIPE_DEFAULTS.k_value
-        self.alpha_value = PIPE_DEFAULTS.alpha_value
+        self.length = length
+        self.diameter = inner_diameter
```

Comment:
One final suggestion, use a more explicit description. You are here actually referring to the inner_diameter so make it known!

Reply:
completed.